crates/matrix-sdk-indexeddb/src/safe_encode.rs

```diff
@@ -39,6 +41,44 @@ pub trait SafeEncode {
         JsValue::from(self.as_encoded_string())
     }
 
+    /// encode self into a JsValue, internally using `as_secure_string`
+    /// to escape the value of self,
+    fn encode_secure(&self, table_name: &str, store_cipher: &StoreCipher) -> JsValue {
+        JsValue::from(self.as_secure_string(table_name, store_cipher))
+    }
+
+    /// encode self securely for the given tablename with the given
+    /// `store_cipher` hash_key, returns the value as a base64 encoded
+    /// string without any padding.
+    fn as_secure_string(&self, table_name: &str, store_cipher: &StoreCipher) -> String {
+        base64_encode(
+            store_cipher.hash_key(table_name, self.as_encoded_string().as_bytes()),
+            STANDARD_NO_PAD,
+        )
+    }
+
+    /// encode self into a JsValue, internally using `as_encoded_string`
+    /// to escape the value of self, and append the given counter
+    fn encode_with_counter(&self, i: usize) -> JsValue {
+        JsValue::from(format!("{}{}{:0000000x}", self.as_encoded_string(), KEY_SEPARATOR, i))
```

Comment:
The `String` > `JsValue` conversion could just be `.into()` since the target of the conversion can be inferred from the function signature.